lib/matplotlib/colors.py

```diff
@@ -62,55 +65,201 @@
 import numpy as np
 from numpy import ma
 import matplotlib.cbook as cbook
-from ._color_data import XKCD_COLORS, CSS4_COLORS
+from ._color_data import BASE_COLORS, CSS4_COLORS, XKCD_COLORS
 
-# for back copatibility
-cnames = CSS4_COLORS
 
-COLOR_NAMES = {'xkcd': XKCD_COLORS,
-               'css4': CSS4_COLORS}
+class _ColorMapping(dict):
+    def __init__(self, mapping):
+        super(_ColorMapping, self).__init__(mapping)
+        self._cache = {}
```

Comment:
Given how this is used, it might make more sense to make `cache` a public attribute.